### DISCUSSION TITLE
Connecting to DSC Alarm panel

### DISCUSSION THREAD

**Discussion by DGoertzen** [arendst/Tasmota#11348] [Ideas]
Has anybody implemented (or tried to) a connection to DSC alarm systems?  [taligentx](https://github.com/taligentx) has created a library for interfacing to esp8266 and esp32 microcontrollers to DSC PowerSeries security systems for integration with home automation systems.  It is available at [dscKeybusInterface](https://github.com/taligentx/dscKeybusInterface).  Has anybody tried to integrate this already?  Thanks!

**Comment by rapi3** [comment]
Thank you very much for share this information.
Can't wait for future integration with DSC or another project that already done this.

**Comment by krishy2** [comment]
DSC integration would be great here!

**Reply by barbudor** [reply]
It requires someone with the device in hands and the will to implement it
Could it be you ?

**Reply by krishy2** [reply]
I own a DSC panel, a Wemos D1 mini is connected, it is working somehow with Talgentix firmware. I am not familiar with Tasmota’s architecture, nor Arduino programming. However I used to code some C once upon a time, and I can read the code. So I won’t be able to do it alone, but with a help of somebody who is familiar with coding it can be done.

**Reply by barbudor** [reply]
I don't know the DSC series of product at all
I see the library is splitted into different parts (Classic, ClassicKeypad, KeybusInterface...)
Does that correspond to different devices ?

IMHO the steps would be
1) What is the purpose to connect it to Tasmota ? What do you want to achieve ?
  => describe use cases
2) What do we need on Tasmota side:
  => New commands ? For configuration only ? For action to the device ?
  => What does the device would report to Tasmota ? How should this be reported to MQTT (ie how would you handle that in the back-end)
3) For each element identified on 2), what calls to the library are required ?

That would help defining the structure of the driver.
Then you mostly have to populate the structure with calls to the library

**Reply by krishy2** [reply]
1. Tasmota would be the "WiFi adapter" between DSC and MQTT or similar.
2. Need to have DSC driver on 2 pins. We need to listen the bus and forward to MQTT what is happening and if we get a MQTT command we should write it on the bus.
DSC will report the state of the zones and partitions, alarm status, system status. We can arm, disarm, acknowledge.
Also a simple web interface would be great.
